content/terminal/custom-backend/Widgets/table/table.md

```diff
@@ -0,0 +1,120 @@
+---
+title: Table Example
+sidebar_position: 1
+description: Learn how to integrate your own backend with OpenBB Terminal Pro using the cookie-cutter or language-agnostic API approaches, with illustrative guides and principles for handling widget.json files, APIs, interfaces, Python, FastAPI, and more.
+keywords:
+- widgets.json
+- OpenBB API
+- Endpoint integration
+- widget configuration
+- Language-Agnostic API
+- API implementation
+- Python
+- FastAPI
+- Terminal Pro widgets
+- Widget definitions
+---
+
+# Introduction
+
+This guide will walk you through the process of creating a table widget for OpenBB Terminal Pro. By the end of this guide, you will have a working table widget that you can add to OpenBB Terminal Pro.
+
+> **Prerequisite:** Ensure you have Python > 3.11 installed on your system before proceeding.
+
+## Step 1: Set Up Your Project
+
+To get started, create the main application file and the widget configuration file. You will only need these two files:
+
+- `main.py`: This file will contain your FastAPI application code.
+- `widgets.json`: This file will define the configuration for your widget.
+
+## Step 2: Create the Backend
+
+The backend will use the same FastAPI setup and structure as described in the [Custom Backend](/content/terminal/custom-backend/custom-backend.md#create-your-own-backend) page.
+
+### Edit the Main Application File
+
+Edit the `main.py` file and add the following code. This code sets up an endpoint to serve the widget configuration, in this case we are using the api.llama.fi/v2/chains endpoint to fetch the data for the table :
+
+```python
+...
+# Table endpoint
+@app.get("/chains_table")
+def chains_table():
+    """Endpoint to fetch the current TVL of all chains using Defi Llama."""
+    params = {}
+    response = requests.get("https://api.llama.fi/v2/chains", params=params)
+
+    if response.status_code == 200:
+        return response.json()
+
+    print(f"Request error {response.status_code}: {response.text}")
+    return JSONResponse(
+        content={"error": response.text}, status_code=response.status_code
+    )
+```
+
+### Edit the widgets.json File
+
+Open the `widgets.json` file and add the following JSON data. This configuration defines the widget's properties and how it should be displayed in the OpenBB Terminal Pro.
+
+```json
+{
+  "chains_table": {
+    "name": "Chains table example",
+    "description": "Get current TVL of all chains using Defi LLama",
+    "category": "crypto",
+    "searchCategory": "crypto",
+    "endpoint": "chains_table",
+    "gridData": {
+      "w": 20,
+      "h": 9
+    },
+    "data": {
+      "table": {
+        "showAll": true,
+        "columnsDefs": [
```

Comment:
Specifying columnDefs explicitly in all tutorials makes them a bit too verbose. Would it make sense to have a tutorial where the bare minimum will be specified and all optional parameters will be omitted?

Reply:
Ya I can take it out here - and point them to the column and cell page